src/Settings.py

```diff
@@ -74,3 +74,38 @@ def get_username(capitalize=False):
             username = username.capitalize()
 
         return username
+
+    def check_discord_endpoints():
+        settings = Settings.get_all()
+
+        if not Settings.discord_prod_key in settings or \
+           not Settings.discord_dev_key in settings:
+            return False
+
+        if settings[Settings.discord_prod_key] == "" or \
+           settings[Settings.discord_dev_key] == "":
+            return False
+
+        return True
+
+    def set_discord_endpoints():
+        if not Settings.check_discord_endpoints():
+            drive = Drive()
+
+            Settings.set_key(
+                key  = Settings.discord_prod_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_prod_key
+                )
+            )
+
+            Settings.set_key(
+                key  = Settings.discord_dev_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_dev_key
+                )
+            )
```

Comment:
@vindennl48 I couldn't really find a better place for all of this, but it seems like it makes more sense here than in the `Discord` class, I think? I'm thinking that this code really just runs once when the program initializes to set some state—maybe we could extract an `Init.py` and have it contain all the setup code?

Reply:
I would disagree. In my opinion, this group of methods does not fall under the Discord umbrella. It falls under the umbrella of setting and verifying variables read from the environment.

However, I would agree that `Settings.py` is not necessarily the right place for it. I would like to see all of this code moved into a `Config.py`, `Env.py`, or similar module. This module should run once at the start of the program. It should set any variables read from `.settings` and verify that all required values are present up front.

I'm happy to start working on this refactoring, but I think the right time to do it would be after this PR has been merged. Would you agree?

Reply:
Not trying to be difficult, but I do not agree.  I think you are bringing a Rails perspective to this when it doesn't necessarily fit (imo).  The last time I refactored this project, I consciously made the decision to have all things that pertain to itself in it's own namespace/class.  I actually had pieces like this spread all over and it became a problem.  Reason being, I don't touch this more than maybe once a year / once every 2 years.  It's much easier to come in and look in the 'Discord' class file and have everything pertaining to discord there instead of having to remember where all the little chunks of discord lay within the project.  

It's setup similar to how the C++ programs work.  Every device/section/topic/(however you want to classify something) is grouped in namespaces or classes.  This way it is easier to find the code in one spot.

Not saying one way is better than the other, just that I haven't used rails in a long time, I'm much more suited to this style and I think it makes more sense specifically for this project because of how little we need to maintain it.